cfme/tests/containers/test_autotagging_providers.py

```diff
@@ -0,0 +1,199 @@
+    # Creating random instance for each object in TEST_OBJECTS and create a random label for it.
+    label_data = namedtuple('label_data', ['instance', 'company_tag',
+                                           'set_value', 'category', 'result_status',
+                                           'result_message', 'label_name'])
+    data_collection = []
+    # Collected data in the form:
+    #                <instance>, <company_tag>, <results_status>
+    # Adding company tags to each object:
+    for test_obj in TEST_OBJECTS:
+        if not label_name:
+            label_name = 'label_' + fauxfactory.gen_alpha(5)
+        get_random_kwargs = {'count': 1, 'appliance': appliance}
+        if test_obj is Image:
+            get_random_kwargs['ocp_only'] = True
+        instance = test_obj.get_random_instances(provider, **get_random_kwargs).pop()
+        set_value = test_obj.__name__ + "_" + value
+        category = 'category' + "_" + fauxfactory.gen_alpha(5)
+        company_tag = category + ': ' + set_value
+        result_status, result_message = set_label(test_obj, instance, label_name, set_value)
+        data_collection.append(
+            label_data(instance, company_tag, set_value, category,
+                       result_status, result_message, label_name)
+        )
+    return data_collection
+
+
+def add_2_labels_same_key_same_value(provider, appliance, value):
+    label_data = namedtuple('label_data', ['instance', 'company_tag',
+                                           'set_value', 'category', 'result_status',
+                                           'result_message', 'label_name'])
+    data_collection1 = []
+    data_collection2 = []
+    first_category = 'category' + "_" + fauxfactory.gen_alpha(5)
+    label_name = 'label_' + fauxfactory.gen_alpha(5)
+    for test_obj in TEST_OBJECTS:
+        get_random_kwargs = {'count': 1, 'appliance': appliance}
+        if test_obj is Image:
+            get_random_kwargs['ocp_only'] = True
+        instance = test_obj.get_random_instances(provider, **get_random_kwargs).pop()
+        category = first_category + "_" + test_obj.__name__
+        company_tag = category + ': ' + value
+        result_status1, result_message1 = set_label(test_obj, instance, label_name, value)
+        instance2 = test_obj.get_random_instances(provider, **get_random_kwargs).pop()
+        while instance2 == instance:
+            instance2 = test_obj.get_random_instances(provider, **get_random_kwargs).pop()
+        result_status2, result_message2 = set_label(test_obj, instance2, label_name, value)
+        data_collection1.append(
+            label_data(instance, company_tag, value, category, result_status1,
+                       result_message1, label_name))
+        data_collection2.append(
+            label_data(instance, company_tag, value, category, result_status2,
+                       result_message2, label_name))
+    return data_collection1, data_collection2
```

Comment:
This function is a mess, must be rewritten.

Reply:
@quarckster please give more info here